app/service/file_svc.py

```diff
@@ -34,30 +34,41 @@ def __init__(self, plugins, exfil_dir):
         except Exception as e:
             return web.HTTPNotFound(body=e)
 
-    async def upload(self, request):
+    async def upload(self, request, file_target=None, filebase=None, xored=False):
         """
         Accept a multipart file via HTTP and save it to the server
-        :param request:
-        :return: None
+        :param request: request object from aiohttp
+        :param file_target: filename of the file to be saved
+        :param filebase: base directory to save an uploaded file to
+        :param xored: whether the file needs to be encrypted on disk or not
+        :return: raw web response for failure or success
         """
         try:
             reader = await request.multipart()
-            exfil_dir = await self._create_exfil_sub_directory(request.headers)
+            save_dir = await self._create_exfil_sub_directory(request.headers)
             while True:
                 field = await reader.next()
                 if not field:
                     break
                 filename = field.filename
-                with open(os.path.join(exfil_dir, filename), 'wb') as f:
+                if file_target:
+                    save_dir = filebase
+                    filename = file_target
+                with open(os.path.join(save_dir, filename), 'wb') as f:
                     while True:
                         chunk = await field.read_chunk()
                         if not chunk:
                             break
                         f.write(chunk)
+                if file_target:
```

Comment:
we're doing 2 checks against file_target == None in this function. we're applying some spaghetti code here, with the multiple if/conditionals. we should look to tighten this up.